sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -843,6 +854,12 @@ def __init__(self, job, runner):
     self._runner = runner
     self.metric_results = None
 
+  def _update_job(self):
+    # We need the job id to be able to update job information. There is no need
+    # to update the job if we are in a known terminal state.
+    if self.has_job and not self._is_in_terminal_state():
+      self._job = self._runner.dataflow_client.get_job(self.job_id)
```

Comment:
<!--thread_id:cc_135126829_t; commit:5b7323ad9e10a5693ea41a239c8a675139b977de; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> This should be `self.job_id()` since the method is not a `@property`.

<!--section:body-->
Done.